packages/seda-chain-contracts/src/contract.rs

```diff
@@ -39,12 +39,37 @@ pub fn execute(
     match msg {
         ExecuteMsg::PostDataRequest {
             dr_id,
-            value,
-            nonce,
-            chain_id,
-        } => data_requests::post_data_request(deps, info, dr_id, value, nonce, chain_id),
-        ExecuteMsg::PostDataResult { dr_id, result } => {
-            data_request_results::post_data_result(deps, info, dr_id, result)
+
+            dr_binary_id,
+            tally_binary_id,
+            dr_inputs,
+            tally_inputs,
+            memo,
+            replication_factor,
+
+            gas_price,
+            gas_limit,
+
+            payload,
+        } => data_requests::post_data_request(
+            deps,
+            info,
+            dr_id,
+            dr_binary_id,
+            tally_binary_id,
+            dr_inputs,
+            tally_inputs,
+            memo,
+            replication_factor,
+            gas_price,
+            gas_limit,
+            payload,
+        ),
```

Comment:
Maybe instead of deconstructing the `ExecuteMsg::PostDataRequest` and constructing it again, we can pass in the struct immediately.
```rust
ExecuteMsg::PostDataRequest(dr) => data_requests::post_data_request(deps, info, dr)
```

Reply:
I would look prettier, right? :)